spaghetti-core/src/main/java/com/prezi/spaghetti/ast/internal/DefaultEnumNode.java

```diff
@@ -48,4 +64,65 @@ public void setDocumentation(DocumentationNode documentation) {
 	public NamedNodeSetInternal<EnumValueNode> getValues() {
 		return values;
 	}
+
+	@Override
+	public NamedNodeSetInternal<EnumValueNode> getNormalizedValues() {
+		NamedNodeSetInternal<EnumValueNode> nodes = makeEmptyEnumValueNodeSet();
+		switch (this.detectAssignmentStyle()) {
+			case NoEntries:
+				break;
+			case Implicit:
+				// Enum entry values are determined from order of appearance
+				int ordinal = 0;
+				for (EnumValueNode enumValueNode : values) {
+					nodes.addInternal(makeValueNodeFrom(enumValueNode, ordinal++));
+				}
+				break;
+			case Explicit:
+				// Enum entry values are explicitly given
+				Set<Integer> seenValues = Sets.newLinkedHashSet();
+				for (EnumValueNode enumValueNode : values) {
+					Integer value = enumValueNode.getValue();
+					if (!seenValues.add(value)) {
+						throw new AstParserException(
+								this.getLocation().getSource(),
+								"Duplicate value in enum " + this.getName());
+					}
+					nodes.addInternal(makeValueNodeFrom(enumValueNode, value));
+				}
+				break;
+			case Mixed:
+				throw new AstParserException(
+						this.getLocation().getSource(),
+						"Mixed implicit and explicit entries in enum " + this.getName());
+		}
+		return nodes;
+	}
```

Comment:
Let's do this in the parser, and get rid of the dynamic calculation and the normalized/non-normalized thing.